### DISCUSSION TITLE
How to detect client (pushState) routing in `loader`?

### DISCUSSION THREAD

**Discussion by maranomynet** [remix-run/remix#5583] [Q&A] [answered]
What is the best way to differentiate between client-routed requests and normal browser/curl requests, within `loader` (and `action`)?

Is this even supported?

* I notice that `loader` seems to receive a request with the `?_data=` parameter **stripped** from the URL.

* I know it's possible to [hijack/overload the global fetch method](https://github.com/remix-run/remix/discussions/2223#discussioncomment-2311081), but that feels like an act of desperatation.

**Rationale:**

I've been thinking of ways to use `defer` but mitigate the issue of HTML arriving out of order, depending on JavaScript to eventually reach correct form.

On initial/full page loads (via outside links, or browser page refresh) I'd like to await all the promises and return an orderly rendered HTML page.

However, when users navigate between pages via Remix's client-side routing, I'd like to return `defer`ed loader data to do instantaneous transition the new UI in a loading state (I know, ick spinners, but hey), and then re-render when the data arrives.

**Comment by Markyiptw** [comment]
#5084 seems to be a viable solution.

**Reply by maranomynet** [reply]
Thank you!
Yes, checking the `Accept` header sounds sufficiently "clean" for my taste.

**Reply by maranomynet** [reply, marked as answer]
Actually it turns out the `Sec-Fetch-Dest` header is the way to go. See [this comment from @sergiodxa](https://github.com/sergiodxa/remix-utils/discussions/311#discussioncomment-8572497)
